internal/config/config_test.go

```diff
@@ -124,3 +127,28 @@ func TestLoad_WithoutConfigFile_UsesDefaults(t *testing.T) {
 		)
 	}
 }
+
+func TestLoad_WithDevModeConfig(t *testing.T) {
+	resetGlobalConfig()
+
+	// Test with dev mode in config file
+	yamlContent := `
+devMode: true
+network: "preview"
+`
+	tmpFile := "test-dev-mode.yaml"
+	err := os.WriteFile(tmpFile, []byte(yamlContent), 0644)
```

Comment:
We should use a temp directory for this test file. This avoids the possibility of accidentally overwriting/removing a file that already exists.